### DISCUSSION TITLE
Rename `commit` cmd --no-commit options to `cache` and --no-cache

### DISCUSSION THREAD

**Discussion by jorgeorpinel** [treeverse/dvc#5948] [New Features & Ideas]
After reviewing and simplifying the `dvc commit` cmd ref (in iterative/dvc.org/pull/1978) I realized it can be confusing to call this "committing" since we're mainly talking about caching (or not) data. Yes, it can also change hash values in DVC files but that's kind of an edge case given the main use for this: Data/code dev process where `add/repro/run --no-commit` are used until ready, and finished by a `commit` — the hash values are already saves in the DVC files by `add/repro/run`.

And we also have `run --metrics/plots-no-cache`, so `no-cache` is more consistent with those.

Finally, changing `dvc commit` to `dvc cache` and `--no-commit` to `--no-cache` would also help avoid confusions related to the Git analogies we use. Specifically, it will help clarify that DVC does not perform the actual versioning, but that Git does, through commits. Currently we need to have the following clarification near the top of the ref:

> Note that this does not register project versions. That can be done with `git commit`.

---

Maybe a change for 2.x? And it doesn't have to break backward compatibility if the old names are kept as aliases.

**Comment by skshetry** [comment]
Magic number :tada: 
![Screenshot from 2020-12-01 11-11-50](https://user-images.githubusercontent.com/18718008/100700625-024bdf00-33c6-11eb-8bc0-fb1831c4e6be.png)
